### PR TITLE
ZRLE implementation

### PR DESCRIPTION
**Pull request by c001c0de**
I have implemented ZRLE encoding. Since this is the most efficient and bandwith saving encoding of all available encodings so far, I would suggest making this the default.

This was originally written under Python 3. I have packported to Python2 using the `ord()` compatibility hack previously used to make hextile encoding work under Python3.

### CODE REVIEW

File: vncdotool/rfb.py [hunk 1]
```diff
@@ -112,6 +116,49 @@
 KEY_SpaceBar=   0x0020
 
 
+# ZRLE helpers
+def _next_bit(it, pixels_in_tile):
```
Comment (sibson) [inline vncdotool/rfb.py]: minor nit, should these have shared ```_zrle``` prefix to more easily identity them?

File: vncdotool/client.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ class AuthenticationError(Exception):
 
 
 class VNCDoToolClient(rfb.RFBClient):
-    encoding = rfb.RAW_ENCODING
+    encoding = rfb.ZRLE_ENCODING
```
Comment (sibson) [inline vncdotool/client.py]: I'm on the fence about changing the default.  I tend to agree with your logic but I'm not sure what general compatibility of servers is with ZRLE and worry about breaking existing users.  I see a couple of options I outline below and am open to others. 

The quickest path to merging this PR would be to keep the default as RAW.  You could then expose a command line option to control the selection of encoding in a follow up PR.  Finally, in a major version bump we can change the default from RAW to ZRLE.  

Another option is I hold this PR up until a major version bump is approaching and then add the CLI switch defaulting to ZRLE.

Comment (c001c0de) [reply vncdotool/client.py]: I have reverted the change of the default encoding

File: vncdotool/rfb.py (hunk 1 above)
Comment (c001c0de) [reply vncdotool/rfb.py]: done

**Comment by sibson** [comment]
Thank you, this is great.

**Pull request merged**